api/src/clojure/org/akvo/flow_api/unilog/unilog.clj

```diff
@@ -154,22 +156,25 @@
     (let [ds (DatastoreServiceFactory/getDatastoreService)
           events (process-new-events (read-event-log db offset))
           user-id (user/id-by-email-or-throw-error remote-api instance-id email)
-          authorized-forms (set (map ds/id (su/list-forms user-id)))
-          authorized-forms-to-load (set/intersection authorized-forms (:forms-to-load events))
+          authorized-forms (set (map ds/id (su/list-forms user-id (:forms-to-load events))))
           form-id->form (reduce (fn [acc form-id]
                                   (assoc acc form-id (su/get-form-definition (long form-id)
                                                                              {:include-survey-id? true})))
                                 {}
-                                authorized-forms-to-load)
-          authorized-surveys (set (map ds/id (su/list* user-id)))
-          authorized-events (filter-events-by-authorization events form-id->form authorized-forms authorized-surveys)
+                                authorized-forms)
+          authorized-surveys (su/list* user-id (:survey-changed events))
```

Comment:
There is no `:survey-changed` key on the `process-new-events` result, but a `::survey-changed`, so  `authorized-surveys` always contains all the surveys the user can see. 

Surveys+Datapoints should follow a flow similar to the Form+FormInstances: `process-new-events` calculates the surveys to be loaded/permissioned, `filter-events-by-authorization` does the filtering/enriching logic.

Test happens to pass because **by mistake we pass a null here and end up getting all surveys**.